Contributing guide

operatingops opened the pull request:
@ralph-tice I like the guide you linked, nice and simple. I love the link about commit messages, too. I read that back when I first started coding and it has served me super well.

I adapted the pattern to work with Python and added a couple personal touches like an issue guide and a different style guide (aka my style guide that I just now published because I finally had a need for it). Want to take a quick look? I haven't done one of these before, all feedback welcome.

Also, do you think it's similar enough that a credit line to Thoughtbot is worthwhile?

ralph-tice commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,48 @@
+The issue tracker is for project changes. Bugfixes, improvements, etc. If you need help, check out
+[operatingops.org](operatingops.org) instead of submitting an issue.
+
+1. Write everything you know. Think of all the questions a maintainer might ask and write in the answers.
```
I'd add something like this:
```The minimum viable set for an actionable issue is:
1) What you expected to happen, and why
2) What happened that was different from your expectations, and why, if you have any theories.
3) Whether or not you can contribute a fix.```

ralph-tice commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,48 @@
+
+1. Write everything you know. Think of all the questions a maintainer might ask and write in the answers.
+
+1. Remember that this is free software! We have day jobs 👷. We might not respond right away.
```
I might instead say something like (but my version here scans too complicated to me...):
```Remember that while this is free software and we have day jobs, that means you get the SLA you pay for and we have less tolerance for rushing in a hacked fix than the right solution.```

ralph-tice commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,48 @@
+1. Make sure the tests pass.
+
+   ```shell
+   tox
````
I get a pair of warnings when I do this after pip installing
```WARNING:test command found but not installed in testenv
  cmd: /usr/local/bin/coverage
  env: /Users/rtice/personal/python_prod_script_pattern/.tox/py36
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.```

operatingops replied:
Reproduced. I'm an idiot and was running in a stale environment that magically worked because old stuff was still installed. There's a clean fix, though, which I'll push in a moment.

ralph-tice commented:
I always like leaving breadcrumbs for derivative works, even if they were just the inspiration when I had an idea, to let someone know what kind of ideas went into the process of creation.

Nice work man!  Wonder if @rocktavious has any input.

This repo is pretty timely, I've had this stuff bubbling around in my head for awhile.

operatingops commented:
Great feedback @ralph-tice! I incorporated your notes and fixed that bug, and also made things look a bit more like the Thoughtbot copy. There ended up being some duplication between readme and contributing guide, and some sloppy differences in linking/line wrapping on my part was really starting to bug me, so I also redid the docs a bit to clean up. Might be easier to read the branche's md files than the diff.

ralph-tice commented:
LGTM! :shipit:

operatingops commented:
Shipped! Thanks for the review @ralph-tice

rocktavious commented:
Sorry just seeing this now - but only thing i would say is that in leu of virtualenv for things that are pip installable and use entrypoints - https://github.com/mitsuhiko/pipsi - is a friggin' godsend!
I never leave home without it!
